xsuite/src/proxy/csproxy.ts

```diff
@@ -0,0 +1,131 @@
+import { addressLikeToBechAddress } from "../data/addressLike";
+import { e, eCodeMetadata, EncodableAccount } from "../data/encoding";
+import { Kvs } from "../data/kvs";
+import { BroadTx, Proxy } from "./proxy";
+
+export class CSProxy extends Proxy {
+  autoGenerateBlocks: boolean;
+  waitCompletedTimeout: number;
+
+  constructor(params: CSProxyParams) {
+    super(params);
+
+    this.autoGenerateBlocks = params.autoGenerateBlocks;
+    this.waitCompletedTimeout = params.waitCompletedTimeout ?? 250;
+  }
+
+  async setAccount(account: EncodableAccount) {
+    const previousKvs = await this.getAccountKvs(account.address);
+    const newAccount = accountToRawAccount(account, previousKvs);
+
+    const result = this.fetch("/simulator/set-state", [newAccount]);
+
+    if (this.autoGenerateBlocks) {
+      await result;
+
+      await this.generateBlock();
+    }
+
+    return result;
+  }
+
+  async sendTx(tx: BroadTx) {
+    const result = super.sendTx(tx);
+
+    if (this.autoGenerateBlocks) {
```

Comment:
This logic should be in CSWorld. The purpose of CSProxy should just be to simplify fetching the chain simulator, not to have chain logic.

Reply:
This is called by the `sendTx` that is not even a method of the World so it can be overwritten easily.

`sendTx` is then also called in World and in `executeTx` which is again not a function of World and is used in multiple other functions that do not belong in World,. like `transfer`, `callContract` etc

This is the only place this code can stay and I don't see any issue with having chain logic in the CSProxy since it is related to the chain simulator only.